src/Mollie/WC/Gateway/Abstract.php

```diff
@@ -2136,4 +2141,110 @@ protected function appendOrderArgumentsToUrl($order_id, $order_key, $webhook_url
         );
         return $webhook_url;
 }
+
+    /**
+     * @param $newOrderStatus
+     *
+     * @param $statusTypeString
+     *
+     * @return mixed|void|null
+     */
+    protected function newStatusFilters($newOrderStatus, $statusTypeString)
+    {
+        // Overwrite plugin-wide
+        $newOrderStatus = apply_filters(
+            Mollie_WC_Plugin::PLUGIN_ID . $statusTypeString,
+            $newOrderStatus
+        );
+
+        // Overwrite gateway-wide
+        $newOrderStatus = apply_filters(
+            Mollie_WC_Plugin::PLUGIN_ID . $statusTypeString . $this->id,
+            $newOrderStatus
+        );
+        return $newOrderStatus;
+    }
+
+    /**
+     * @param Mollie\Api\Resources\Payment|Mollie\Api\Resources\Order $payment
+     *
+     * @return bool
+     */
+    protected function isPartialRefund($payment)
+    {
+        return ($payment->amount->value - $payment->amountRefunded->value) !== 0;
+    }
+
+    /**
+     * @param WC_Order                                                $order
+     * @param Mollie\Api\Resources\Payment|Mollie\Api\Resources\Order $payment
+     */
+    protected function processUpdateStateRefund(WC_Order $order, $payment)
+    {
+        $this->isPartialRefund($payment)
+            ? $this->updateStateRefund(
+            $order,
+            $payment,
+            self::STATUS_ON_HOLD,
+            '_order_status_partially_refunded'
+        )
+            : $this->updateStateRefund(
+            $order,
+            $payment,
+            self::STATUS_REFUNDED,
+            '_order_status_refunded');
+    }
+
+    /**
+     * @param WC_Order                                                $order
+     * @param Mollie\Api\Resources\Payment|Mollie\Api\Resources\Order $payment
+     * @param                                                         $newOrderStatus
+     * @param                                                         $statusFilterString
+     */
+    protected function updateStateRefund(
+        WC_Order $order,
+        $payment,
+        $newOrderStatus,
+        $statusFilterString
+    ) {
+        // New order status
+        $newOrderStatus = $this->newStatusFilters(
+            $newOrderStatus,
+            $statusFilterString
+        );
+        $note = $this->notePrint($payment, $newOrderStatus);
+        $this->updateOrderStatus(
+            $order,
+            $newOrderStatus,
+            $note,
+            $restoreStock = false
+        );
+    }
+
+    /**
+     * @param $payment
+     * @param $newOrderStatus
+     *
+     * @return string
+     */
+    protected function notePrint($payment, $newOrderStatus)
```

Comment:
The method does not print anything, may be you want to call it `renderNote` because it render does not print.